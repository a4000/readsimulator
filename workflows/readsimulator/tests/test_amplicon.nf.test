nextflow_workflow {

    name "Test workflow: READSIMULATOR"
    script "../main.nf"
    workflow "READSIMULATOR"
    tag "workflows"
    tag "readsimulator"
    tag "readsimulator_test_amplicon"

    test("amplicon = true") {

        when {
            workflow {
                """
                input[0] = Channel.of(
                    [ [ id:'first', seed:40] ],
                    [ [ id:'second', seed:41] ],
                    [ [ id:'third', seed:42] ],
                )
                """
            }
            params {
                amplicon                    = true
                fw_primer                   = 'AAAATAAT'
                rv_primer                   = 'GATTACTTT'
                amplicon_read_count         = 1000
                amplicon_crabs_ispcr_error  = 0
                fasta                       = 'https://raw.githubusercontent.com/nf-core/test-datasets/readsimulator/testdata/GCF_024334085.1_ASM2433408v1_genomic.fna.gz'
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out
                ).match() }
            )
        }
    }
}
