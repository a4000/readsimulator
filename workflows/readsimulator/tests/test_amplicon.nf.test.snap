{
    "amplicon = true": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "first",
                            "seed": 40,
                            "outdir": "art_illumina",
                            "datatype": "amplicon_illumina"
                        },
                        [
                            "first1.fq.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "first2.fq.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ],
                    [
                        {
                            "id": "second",
                            "seed": 41,
                            "outdir": "art_illumina",
                            "datatype": "amplicon_illumina"
                        },
                        [
                            "second1.fq.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "second2.fq.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ],
                    [
                        {
                            "id": "third",
                            "seed": 42,
                            "outdir": "art_illumina",
                            "datatype": "amplicon_illumina"
                        },
                        [
                            "third1.fq.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "third2.fq.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "amplicon_illumina"
                        },
                        "amplicon_illumina_samplesheet.csv:md5,2f0440b621dd2de3c89cda85c81e5be7"
                    ]
                ],
                "samplesheet": [
                    [
                        {
                            "id": "amplicon_illumina"
                        },
                        "amplicon_illumina_samplesheet.csv:md5,2f0440b621dd2de3c89cda85c81e5be7"
                    ]
                ],
                "simulated_reads": [
                    [
                        {
                            "id": "first",
                            "seed": 40,
                            "outdir": "art_illumina",
                            "datatype": "amplicon_illumina"
                        },
                        [
                            "first1.fq.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "first2.fq.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ],
                    [
                        {
                            "id": "second",
                            "seed": 41,
                            "outdir": "art_illumina",
                            "datatype": "amplicon_illumina"
                        },
                        [
                            "second1.fq.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "second2.fq.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ],
                    [
                        {
                            "id": "third",
                            "seed": 42,
                            "outdir": "art_illumina",
                            "datatype": "amplicon_illumina"
                        },
                        [
                            "third1.fq.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "third2.fq.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ]
            }
        ],
        "timestamp": "2024-01-19T19:15:08.881730553"
    }
}